src/data/nat/prime.lean

```diff
@@ -541,20 +541,7 @@ by rw [←prime_iff, prime]
 /-- Prime `p` divides the product of `L : list ℕ` iff it divides some `a ∈ L` -/
 lemma prime.dvd_prod_iff {p : ℕ} {L : list ℕ} (pp : p.prime) :
   p ∣ L.prod ↔ ∃ a ∈ L, p ∣ a :=
-begin
-  split,
-  { intros h,
-    induction L,
-    { simp only [list.prod_nil] at h, exact absurd h (prime.not_dvd_one pp) },
-    { rw list.prod_cons at h,
-      cases (prime.dvd_mul pp).mp h,
-      { use L_hd, simp [h_1] },
-      { rcases L_ih h_1 with ⟨x, hx1, hx2⟩, use x, simp [list.mem_cons_iff, hx1, hx2] } } },
-  { exact λ ⟨a, ha1, ha2⟩, dvd_trans ha2 (list.dvd_prod ha1) },
-end
--- TODO: This proof duplicates a more general proof in `algebra/associated`.
--- The two proofs should be integrated after the merger of `nat.prime` and `prime`
--- that's about to occur. (2021-12-17)
+_root_.prime.dvd_prod_iff $ prime_iff.mp pp
```

Comment:
This proof is now so short that one could almost wonder if the lemma needs to be kept around at all.
One reason why I wanted to replace the old definition of `nat.prime` was that we could probably deduplicate a lot of API.